src/types/array.ts

```diff
@@ -21,17 +21,20 @@ function InternalArr<E extends Runtype, RO extends boolean>(
 ): Arr<E, RO> {
   return withExtraModifierFuncs(
     create(
-      xs => {
+      (xs, visitedSet = new Set(), failedSet = new Set()) => {
```

Comment:
Shouldn't need to create defaults here, all arguments should be supplied for you.

Reply:
I'm not sure I understand. How can we know that the defaults will be supplied? for instance, in the graph test case wouldn't this validate function be called at the top level?

Reply:
This is not the `validate` function, this is the argument to `create`, which is an internal utility. We can guarantee that the function passed to `create` always has its arguments supplied.

Reply:
Ok, so we would modify the create function in `runtypes.ts` to wrap the passed in function in something that provides it with empty sets and assign the wrapping function to `validate`?

Reply:
I think 77eed40 should address this, if I've understood you correctly.